### PR TITLE
Allow for deployment of puppet 8. Fixes #36939

### PR DESCRIPTION
**Pull request by trenta**
Modified templates in unattended/provisioning_templates adding the option to deploy with puppet 8.

### CODE REVIEW

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb [hunk 1]
```diff
@@ -336,7 +337,7 @@ chmod +x /root/remote_execution_pull_setup.sh
 <% end -%>
 
 <% if puppet_enabled %>
-<% if host_param_true?('enable-puppetlabs-repo') || host_param_true?('enable-official-puppet7-repo') || host_param_true?('enable-puppetlabs-puppet6-repo')|| host_param_true?('enable-puppetlabs-puppet5-repo') -%>
+<% if host_param_true?('enable-puppetlabs-repo') ||host_param_true?('enable-official-puppet8-repo') || host_param_true?('enable-official-puppet7-repo') || host_param_true?('enable-puppetlabs-puppet6-repo')|| host_param_true?('enable-puppetlabs-puppet5-repo') -%>
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/provision/kickstart_default.erb]:
```suggestion
<% if host_param_true?('enable-puppetlabs-repo') || host_param_true?('enable-official-puppet8-repo') || host_param_true?('enable-official-puppet7-repo') || host_param_true?('enable-puppetlabs-puppet6-repo')|| host_param_true?('enable-puppetlabs-puppet5-repo') -%>
```

File: app/views/unattended/provisioning_templates/user_data/autoyast_default_user_data.erb [hunk 2]
```diff
@@ -31,7 +31,7 @@ echo 'root:<%= root_pass -%>' | /usr/sbin/chpasswd -e
 <%= snippet "blacklist_kernel_modules" %>
 
 <% if puppet_enabled %>
-<% if host_param_true?('enable-official-puppet7-repo') || host_param_true?('enable-puppetlabs-puppet6-repo') || host_param_true?('enable-puppetlabs-puppet5-repo') -%>
+<% if  host_param_true?('enable-official-puppet8-repo') || host_param_true?('enable-official-puppet7-repo') || host_param_true?('enable-puppetlabs-puppet6-repo') || host_param_true?('enable-puppetlabs-puppet5-repo') -%>
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/user_data/autoyast_default_user_data.erb]:
```suggestion
<% if host_param_true?('enable-official-puppet8-repo') || host_param_true?('enable-official-puppet7-repo') || host_param_true?('enable-puppetlabs-puppet6-repo') || host_param_true?('enable-puppetlabs-puppet5-repo') -%>
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Thanks for this. Small whitespace suggestions, but otherwise it looks good to me.

**Comment by trenta** [comment]
Do you need me to fix the white spaces or will the person doing the merge do it then?

**Comment by ekohl** [comment]
Please fix up the commit. We had too many people who used the squash function wrong that we've disabled it.

**Comment by trenta** [comment]
I get that. I'm probably one of those people. My git skills leave a lot to be desired. I've pushed the changes.

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
ok to test

**Pull request merged**